### PR TITLE
Logging map

### PR DESCRIPTION
**Pull request by kartikeytewari-ul**
## Test Plan
1. Issue-98 is fixed
- Link: https://drive.google.com/file/d/1Z9u-r-BvazMSV_PaJGszw9UGW1IJ2eGZ/view?usp=sharing

2. Compile Pipeline
- CI is passing

3. Replay Test (maven-demo java-17)
- SDK 0.6.7
Screenshot 2024-07-19 at 7 09 46 PM

- SDK 0.6.8
Screenshot 2024-07-19 at 7 08 55 PM

4. Replay Test (maven-demo java-8)
- SDK 0.6.7
Screenshot 2024-07-19 at 7 18 47 PM

- SDK 0.6.8
Screenshot 2024-07-19 at 7 19 29 PM

5. Webflux-demo (java-8)
- SDK 0.6.7
Screenshot 2024-07-19 at 7 25 48 PM

- SDK 0.6.8
Screenshot 2024-07-19 at 7 26 17 PM

6. Webflux-demo (java-11)
- SDK 0.6.7
Screenshot 2024-07-19 at 7 28 04 PM

- SDK 0.6.8
Screenshot 2024-07-19 at 7 28 29 PM

7. Auto Executor (maven-demo)
- SDK 0.6.7
```
-----Test Summary-----

    Integration Mode ->
    Total number of cases run : 442
    Number of Passing cases : 416
    Number of Failing cases : 26
    Failing tests (Row numbers) : [50, 51, 53, 54, 93, 94, 100, 122, 123, 124, 125, 130, 131, 132, 133, 178, 179, 185, 186, 195, 196, 345, 346, 347, 352, 354]
    Unit Mode ->
    Total number of cases run : 334
    Number of Passing cases : 320
    Number of Failing cases : 14
    Failing tests (Row numbers) : [28, 29, 31, 32, 100, 101, 110, 111, 163, 164, 173, 174, 319, 335]
```

- SDK 0.6.8
```
  Integration Mode ->
    Total number of cases run : 442
    Number of Passing cases : 416
    Number of Failing cases : 26
    Failing tests (Row numbers) : [50, 51, 53, 54, 93, 94, 100, 122, 123, 124, 125, 130, 131, 132, 133, 178, 179, 185, 186, 195, 196, 345, 346, 347, 352, 354]
    Unit Mode ->
    Total number of cases run : 334
    Number of Passing cases : 320
    Number of Failing cases : 14
    Failing tests (Row numbers) : [28, 29, 31, 32, 100, 101, 110, 111, 163, 164, 173, 174, 319, 335]
```

8. Auto Executor (webflux-demo)
- SDK 0.6.7
```
 Integration Mode ->
    Total number of cases run : 626
    Number of Passing cases : 512
    Number of Failing cases : 114
    Failing tests (Row numbers) : [8, 9, 10, 11, 34, 35, 36, 37, 38, 39, 50, 74, 76, 111, 112, 113, 114, 115, 116, 117, 168, 170, 172, 174, 176, 178, 180, 182, 184, 186, 226, 227, 239, 240, 241, 243, 244, 245, 246, 286, 291, 293, 298, 299, 302, 303, 345, 346, 347, 348, 349, 350, 351, 352, 354, 355, 356, 357, 361, 365, 366, 367, 368, 389, 390, 391, 392, 393, 394, 396, 397, 398, 422, 423, 448, 552, 553, 554, 564, 565, 566, 568, 569, 570, 572, 573, 574, 576, 577, 578, 581, 582, 586, 588, 591, 594, 595, 596, 598, 599, 600, 601, 603, 604, 605, 608, 611, 616, 617, 618, 621, 624, 625, 627]
    Unit Mode ->
    Total number of cases run : 474
    Number of Passing cases : 409
    Number of Failing cases : 65
    Failing tests (Row numbers) : [2, 3, 15, 16, 41, 67, 68, 69, 70, 90, 92, 115, 125, 131, 132, 149, 154, 155, 160, 161, 162, 163, 164, 204, 205, 206, 207, 210, 211, 212, 213, 224, 246, 247, 248, 249, 250, 251, 252, 290, 295, 296, 313, 315, 316, 317, 326, 364, 365, 376, 378, 380, 382, 384, 386, 388, 390, 392, 394, 402, 403, 414, 416, 420, 423]
```

- SDK 0.6.8
```
 Integration Mode ->
    Total number of cases run : 626
    Number of Passing cases : 512
    Number of Failing cases : 114
    Failing tests (Row numbers) : [8, 9, 10, 11, 34, 35, 36, 37, 38, 39, 50, 74, 76, 111, 112, 113, 114, 115, 116, 117, 168, 170, 172, 174, 176, 178, 180, 182, 184, 186, 226, 227, 239, 240, 241, 243, 244, 245, 246, 286, 291, 293, 298, 299, 302, 303, 345, 346, 347, 348, 349, 350, 351, 352, 354, 355, 356, 357, 361, 365, 366, 367, 368, 389, 390, 391, 392, 393, 394, 396, 397, 398, 422, 423, 448, 552, 553, 554, 564, 565, 566, 568, 569, 570, 572, 573, 574, 576, 577, 578, 581, 582, 586, 588, 591, 594, 595, 596, 598, 599, 600, 601, 603, 604, 605, 608, 611, 616, 617, 618, 621, 624, 625, 627]
    Unit Mode ->
    Total number of cases run : 474
    Number of Passing cases : 409
    Number of Failing cases : 65
    Failing tests (Row numbers) : [2, 3, 15, 16, 41, 67, 68, 69, 70, 90, 92, 115, 125, 131, 132, 149, 154, 155, 160, 161, 162, 163, 164, 204, 205, 206, 207, 210, 211, 212, 213, 224, 246, 247, 248, 249, 250, 251, 252, 290, 295, 296, 313, 315, 316, 317, 326, 364, 365, 376, 378, 380, 382, 384, 386, 388, 390, 392, 394, 402, 403, 414, 416, 420, 423]
```

### CODE REVIEW

**Comment by TheAmg** [comment]
Webflux-demo cases - Java 21 (Reply)

SKD 0.6.7 and SDK 0.6.8 (run as 0.6.89)

Screenshot 2024-07-19 at 7 33 34 PM

Screenshot 2024-07-19 at 7 35 12 PM
th name 0.6.89) have Identical results.

**Comment by TheAmg** [comment]
SDK Issue #70 doesn't occur on this build.

Screenshot 2024-07-19 at 7 47 14 PM

**Comment by TheAmg** [comment]
Confirmed that Issue #65 is not happening.
Additionally maps for methods with different signatures are also different (#83).

**Comment by TheAmg** [comment]
#69 also doesn't occur.

The naming convention has changed, there is a probed and unprobed pair and lambda is also probed.

```java
[Decompiler.com](https://www.decompiler.com/)
[JAR String Editor](https://www.decompiler.com/jar-string-editor)
[Release Notes](https://www.decompiler.com/release-notes)
[support@decompiler.com](mailto:support@decompiler.com?subject=Bug%20report%2C%20feature%20request%2C%20suggestion.)

CronController.java 
[CronController.java](https://www.decompiler.com/jar/f95a2a0cbcb5491793d041e0d49b74d7/CronController.java)
[Download file](https://www.decompiler.com/download/f95a2a0cbcb5491793d041e0d49b74d7/CronController.java)
    package org.unlogged.springwebfluxdemo.controller;

import io.unlogged.Runtime;
import io.unlogged.logging.Logging;
import java.util.function.Function;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;
import org.unlogged.springwebfluxdemo.service.CronService;
import reactor.core.publisher.Mono;

@RestController
@RequestMapping({"/cron"})
public class CronController {
   private final CronService cronService;

   @Autowired
   public CronController(CronService cronService) {
      byte var2 = 0;
      Logging.recordEvent(15151531);
      Logging.recordEvent(cronService, 15151532);
      Logging.recordEvent(15151533);
      Logging.recordEvent(15151534);
      int var4 = 15151534;
      super();
      var2 = 0;
      Logging.recordEvent(15151535);
      Logging.recordEvent(this, 15151536);

      try {
         Logging.recordEvent(15151537);
         Logging.recordEvent(cronService, 15151538);
         Logging.recordEvent(this, 15151539);
         Logging.recordEvent(cronService, 15151540);
         var4 = 15151539;
         this.cronService = cronService;
         var2 = 0;
         Logging.recordEvent(15151541);
         Logging.recordEvent(15151542);
      } catch (Throwable var3) {
         Logging.recordEvent(var2, 15151543);
         Logging.recordEvent(var3, 15151544);
         Logging.recordEvent(var3, 15151545);
         throw var3;
      }
   }

   public Mono<String> Unlogged$Probed$$org$unlogged$springwebfluxdemo$controller$CronController$cronHello() {
      byte var1 = 0;

      try {
         Logging.recordEvent(this, 15151548);
         Logging.recordEvent(15151549);
         Logging.recordEvent(this, 15151550);
         int var4 = 15151550;
         CronService var10000 = this.cronService;
         var1 = 0;
         Logging.recordEvent(var10000, 15151551);
         Logging.recordEvent(var10000, 15151552);
         var4 = 15151552;
         Mono var5 = var10000.getCount();
         var1 = 0;
         var5 = Logging.recordEvent(var5, 15151553);
         Logging.recordEvent(15151554);
         Function var10001 = (count) -> {
            return Runtime.probeCounter("org.unlogged.springwebfluxdemo.controller.CronController$lambda$cronHello$0$(Ljava/lang/Integer;)Ljava/lang/String;", 1L) ? Unlogged$Probed$$org$unlogged$springwebfluxdemo$controller$CronController$lambda$cronHello$0(count) : "Hello Cron: " + count;
         };
         Logging.recordEvent(var10001, 15151555);
         Function var2 = var10001;
         Logging.recordEvent(var5, 15151556);
         Logging.recordEvent(var2, 15151557);
         var4 = 15151556;
         var5 = var5.map(var2);
         var1 = 0;
         return Logging.recordEvent(Logging.recordEvent(var5, 15151558), 15151559);
      } catch (Throwable var3) {
         Logging.recordEvent(var1, 15151560);
         Logging.recordEvent(var3, 15151561);
         Logging.recordEvent(var3, 15151562);
         throw var3;
      }
   }

   @GetMapping({"/hello"})
   public Mono<String> cronHello() {
      return Runtime.probeCounter("org.unlogged.springwebfluxdemo.controller.CronController$cronHello$()Lreactor/core/publisher/Mono;", 1L) ? this.Unlogged$Probed$$org$unlogged$springwebfluxdemo$controller$CronController$cronHello() : this.cronService.getCount().map((count) -> {
         return Runtime.probeCounter("org.unlogged.springwebfluxdemo.controller.CronController$lambda$cronHello$0$(Ljava/lang/Integer;)Ljava/lang/String;", 1L) ? Unlogged$Probed$$org$unlogged$springwebfluxdemo$controller$CronController$lambda$cronHello$0(count) : "Hello Cron: " + count;
      });
   }

   // $FF: synthetic method
   private static String Unlogged$Probed$$org$unlogged$springwebfluxdemo$controller$CronController$lambda$cronHello$0(Integer count) {
      byte var1 = 0;

      try {
         Logging.recordEvent(15151565);
         Logging.recordEvent(count, 15151566);
         Logging.recordEvent(15151567);
         Logging.recordEvent(count, 15151568);
         Logging.recordEvent(15151569);
         Logging.recordEvent(count, 15151570);
         String var10000 = "Hello Cron: " + count;
         Logging.recordEvent(var10000, 15151571);
         Logging.recordEvent(var10000, 15151572);
         return var10000;
      } catch (Throwable var3) {
         Logging.recordEvent(var1, 15151573);
         Logging.recordEvent(var3, 15151574);
         Logging.recordEvent(var3, 15151575);
         throw var3;
      }
   }

   static {
      Runtime.registerClass("eNrNV1tz20QUdpqQNg3X8lCGF2aYzJAUY11sSxbGHYKbTD04TqcpYRiGh9XuylEjaz2SnDa88Qd4hN/A/X6/3+/l3vLKEz8DdqW1vZFkOQ4ME2eUWek75zufds85u8pNPXYll8tViNcu9FyHtNsYFfyuZ7vtS9i0nN5lhDukAIkbeMRxsFeoe8StD26p7617nxQugh3AKaU+pZSglIaUUoJytkW8DnDoaFG2LF3XTA2UZBlohlWCZVguqbqB1IqOIUK4hIBZKlHblSYBCHv3MwHw3n8Rvv+7mTFJDnDb0rp5EcMgfCrl2JxdP30195fyzN/8xz1uCf8P8WcFtrvpNUOvo43asr/rQsHsOcFsKoXmeQE/woTF8Be4D/NVU/xf5M+mWXg+xnXgOBd2u7h2HkNs72CvRYKGawc2cOzHMco3XD/wejCwiVtrtDbXH1zZOLdSbyw38+uXXOzV4pOTb4EOrt1nU4rT+TPYh7XFpU1BxEuCiDk+vp2JOAc87AY1pcz+iqV8KEr0fFnwnE55/Vfo3Ql6XTfi9V/lOHv1e7i/EUZpjskSH3s7NsRhimxE46pA/BonPsaDMuIno9k5aPrlV23soHAq4TBo/r9Q+7qg9sShV/sGay30Oj5iUd/k+DwvK4bPJlLnLW51A2eaibG8LeDzKan1joDPxvBrFH83owEw/L3xDYCZvR9rAPEwH9C7kyMKnOEfcpxVx42HemmZ2o8EtTcderUfc7UzvIMwtU8JzbPdc4CX0i03G+cvPDTolpNriPppGwd10nODfkdtehjAgHh0KjwsdXumY/tbdDrWiEtE1Z8Ique4ai3Rbcsqn6J9sn7KWVmnXYzycOrP5KufeaS1vNaoR28Aul1ndyA/3DV6ge1IVs8NfaRVPqjmHyAkoFygK2wutrtDtrHUBB0TgTUcACtUujs0pg+3CKp10rAobDNBF/mcBS5ysL/QJGS7160KZhsBW6HqKEc2aQcDo5DZvktJlK3chh1gYY6WvbYvs7wqkHD/pelUAH6nwCj2WikpVpGOvXbqCDYhAT4TEuBUSqP6nOOzQrE8PXGxZGRjlFMd0M2LSzsiofZdLV8Iso9x2aeT1aJR5Qhfrim8arJiC+xfCuwZtahNVotfcdajvMoZ650TMXzNGeZSNkeGfyPg8ymL/a2AxzfHPyj+XcbmyPDvx2+OzOyHjNMxw69knI4Z/mPG5snwnzh+hB8PWQLcJixwWIcNN8Bt7FUFv5+FbazfDJ8Y0ww7YBvTXQuC4GE72KIjPwBu4OcTfaofbynZlLLaZGQSRVgd1ShHaPi/myY3f7QZ/5KYqAEyywKzLER8wgL9IizQEk+M/Szsr4LfqZSE+k3YW/uVdzLOy99OcPtdaAPxcmP4VQGfTwl7TcCH5cZk9s+r9Hn0/UVH+uJBzj30iy0qlBFf9YvQsiDElaJWLAJdKWuGjlVZkyFGclE3LBlAHSJgKYM2QdsDO5GdxY5D6M0d444w2eGLso5UCIsIymrFUPRKSTGKQFZguWxA06SiDF3FAAymjX7qOOHRYWGgYkGmT+/aZ7HRaT6eoQdVTFBWFQChbuCyrBglOjUGAlimk2PKmoUqGigi9R9pCgBz", "eNpjeG64iuG54RogXg/E24B4JxDvBuKDQHwUiI8B8XEgPgXEZ4D4HBBfAeLrADSXG7o=");
      Runtime.registerMethod("org.unlogged.springwebfluxdemo.controller.CronController$lambda$cronHello$0$(Ljava/lang/Integer;)Ljava/lang/String;");
      Runtime.registerMethod("org.unlogged.springwebfluxdemo.controller.CronController$cronHello$()Lreactor/core/publisher/Mono;");
   }
}
    [Download file](https://www.decompiler.com/download/f95a2a0cbcb5491793d041e0d49b74d7/CronController.java)
```

File: src/main/java/io/unlogged/Runtime.java [hunk 1]
```diff
@@ -38,6 +39,8 @@ public class Runtime {
     private IEventLogger logger = Logging.initialiseDiscardLogger();
     private long lastProbesLoadTime;
 
+    private static HashMap<String, Integer> frequencyMap = new HashMap<>();
```
Comment (kartikeytewari-ul) [inline src/main/java/io/unlogged/Runtime.java:42]: This needs to be shifted to string and long

Comment (kartikeytewari-ul) [reply src/main/java/io/unlogged/Runtime.java:42]: Resolved now

**Pull request merged**